src/session/src/context.rs

```diff
@@ -132,6 +128,70 @@ impl QueryContext {
     pub fn set_time_zone(&self, tz: Option<TimeZone>) {
         let _ = self.time_zone.swap(Arc::new(tz));
     }
+
+    #[inline]
+    pub fn trace_id(&self) -> u64 {
+        self.trace_id
+    }
+}
+
+#[derive(Default)]
+pub struct QueryContextBuilder {
```

Comment:
There is a `derive_builder` macro to generate this.

Reply:
`#[derive(Builder)]` requires fields implement `Clone` too, which `ArcSwap` does not. This can be done together when make catalog and schema immutable within QueryContext(then `ArcSwap` would be unnecessary)

Reply:
:+1: